metadata/src/test/java/org/apache/kafka/image/FakeSnapshotWriter.java

```diff
@@ -33,11 +32,7 @@ public class FakeSnapshotWriter implements SnapshotWriter<ApiMessageAndVersion>
     private boolean closed = false;
 
     public List<List<ApiMessageAndVersion>> batches() {
-        List<List<ApiMessageAndVersion>> result = new ArrayList<>();
-        for (List<ApiMessageAndVersion> batch : batches) {
-            result.add(Collections.unmodifiableList(batch));
-        }
-        return Collections.unmodifiableList(result);
+        return new ArrayList<>(batches);
```

Comment:
if we want to follow origin design, all lists in this method should be immutable.
```java
return batches.stream().map(List::copyOf).toList();
```